libp2p-websocket/src/browser.rs

```diff
@@ -206,41 +206,46 @@ impl Transport for BrowserWsConfig {
 	}
 
     fn nat_traversal(&self, server: &Multiaddr, observed: &Multiaddr) -> Option<Multiaddr> {
-        let server_protocols: Vec<_> = server.iter().collect();
-        let observed_protocols: Vec<_> = observed.iter().collect();
+		let mut server_protocols = server.iter();
+		let server_proto0 = if let Some(p) = server_protocols.next() { p } else { return None; };
```

Comment:
latest `stable` (`rustc 1.23.0 (766bd11c8 2018-01-01)` ) supports `?` operator for `Option` :)